core/store/src/trie/shard_tries.rs

```diff
@@ -217,8 +221,10 @@ impl ShardTries {
         fields(ops.len = ops.len()),
     )]
     pub fn update_cache(&self, ops: Vec<(&CryptoHash, Option<&[u8]>)>, shard_uid: ShardUId) {
-        let cache = self.get_trie_cache_for(shard_uid, false);
-        cache.update_cache(ops);
+        if self.get_mem_tries(shard_uid).is_none() {
+            let cache = self.get_trie_cache_for(shard_uid, false);
```

Comment:
I would expect that `get_trie_cache_for` would return `Option<...>` (and None if it was not initialized) and then we would check if it is some instead of memtries is none. It seems the problem with that is, `caches` and `view_caches` will be initialized (to empty) even if memtries is enabled. maybe we can make them Option later?

Reply:
implemented in 586a461a6
let's keep `caches` and `view_caches` as it is for now, I don't expect much overhead from having empty cache for each shard

Reply:
thanks.